online-shop/src/reducers/ProductDetailsReducer.tsx

```diff
@@ -0,0 +1,135 @@
+    GET_PRODUCT_DETAILS_REQUEST,
+    GET_PRODUCT_DETAILS_SUCCESS,
+    GET_PRODUCT_DETAILS_FAIL,
+    DELETE_PRODUCT_REQUEST,
+    DELETE_PRODUCT_SUCCESS,
+    DELETE_PRODUCT_FAIL,
+    CLEAR_DELETE_STATUS} from '../util/ActionTypes';
+import { ProductDetailsAction, SetProductAction, SetLoadingDetailsAction, 
+    GetProductSuccessAction } from '../actions/ProductDetailsActions';
+import { DEFAULT_ID, DEFAULT_NAME, DEFAULT_CATEGORY, DEFAULT_PRICE, 
+    DEFAULT_IMAGE, DEFAULT_DESCRIPTION, STATUS_NONE, STATUS_SUCCESS, STATUS_FAIL } from "../util/util";
+
+export interface ProductDetailsState {
+    product: Product,
+    isLoading: boolean,
+    showModal: boolean,
+    hasFetchError: boolean,
+    deleteStatus: string,
+};
+
+const initialState: ProductDetailsState = {
+    product: new Product(DEFAULT_ID, DEFAULT_NAME, DEFAULT_CATEGORY,
+        DEFAULT_PRICE, DEFAULT_IMAGE, DEFAULT_DESCRIPTION),
+    isLoading: true,
+    showModal: false,
+    hasFetchError: false,
+    deleteStatus: STATUS_NONE,
+}
+
+export const ProductDetailsReducer = (
+            state: ProductDetailsState = initialState, 
+            action: ProductDetailsAction,
+        ): ProductDetailsState => {
+            
+    switch (action.type) {
+        case SET_PRODUCT: {
+            const actualAction: SetProductAction = action as SetProductAction;
+            return {
+                product: actualAction.product,
+                isLoading: state.isLoading,
+                showModal: state.showModal,
+                hasFetchError: state.hasFetchError,
+                deleteStatus: state.deleteStatus,
+            }
+        }
+        case SET_LOADING_STATUS_DETAILS: {
+            const actualAction: SetLoadingDetailsAction = action as SetLoadingDetailsAction;
+            return {
+                product: state.product,
+                isLoading: actualAction.loadingStatus,
+                showModal: state.showModal,
+                hasFetchError: state.hasFetchError,
+                deleteStatus: state.deleteStatus,
+            }
+        }
+        case OPEN_MODAL_DETAILS: {
+            const newState = true;
+            return {
+                product: state.product,
+                isLoading: state.isLoading,
+                showModal: newState,
+                hasFetchError: state.hasFetchError,
+                deleteStatus: state.deleteStatus,
+            }
+        }
+        case CLOSE_MODAL_DETAILS: {
+            const newState = false;
+            return {
+                product: state.product,
+                isLoading: state.isLoading,
+                showModal: newState,
+                hasFetchError: state.hasFetchError,
+                deleteStatus: state.deleteStatus,
+            }
+        }
+        case GET_PRODUCT_DETAILS_REQUEST: {
+            return state;
+        }
+        case GET_PRODUCT_DETAILS_SUCCESS: {
+            const actualAction: GetProductSuccessAction = action as GetProductSuccessAction;
```

Comment:
You are forced to do this cast because of the fact that all your types are simple strings --> if you would use an enum for the action types + classes for the actions, the concrete type would be inferend automatically by typescript from the switch,